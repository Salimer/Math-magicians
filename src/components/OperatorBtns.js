import PropTypes from 'prop-types';

function OperatorBtns({ handleButtonClick }) {
  return (
    <>
    <div className="operator">
      <div className="button" onClick={handleButtonClick}>รท</div>
      <div className="button" onClick={handleButtonClick}>*</div>
      <div className="button" onClick={handleButtonClick}>-</div>
      <div className="button" onClick={handleButtonClick}>+</div>
      <div className="button" onClick={handleButtonClick}>=</div>
    </div>
  );
}

OperatorBtns.propTypes = { handleButtonClick: PropTypes.func.isRequired };

export default OperatorBtns;
