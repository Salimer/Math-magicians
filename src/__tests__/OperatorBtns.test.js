import { render } from '@testing-library/react';
import OperatorBtns from '../components/OperatorBtns';

test('renders operator buttons correctly', () => {
  const handleButtonClick = jest.fn();
  const { container } = render(
    <OperatorBtns handleButtonClick={handleButtonClick} />,
  );
  expect(container).toMatchSnapshot();
});
