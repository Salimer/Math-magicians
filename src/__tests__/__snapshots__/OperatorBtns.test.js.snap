// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders operator buttons correctly 1`] = `
<div>
  <div
    class="operator"
  >
    <button
      class="button"
      type="button"
    >
      รท
    </button>
    <button
      class="button"
      type="button"
    >
      x
    </button>
    <button
      class="button"
      type="button"
    >
      -
    </button>
    <button
      class="button"
      type="button"
    >
      +
    </button>
    <button
      class="button"
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
